### PR TITLE
Allow variable overriding for ec2 deployment_vars

### PR DESCRIPTION
**Pull request by detiber**
- users can now override the deployment_vars variables with the assocated
  ec2_* variables
- added deployment_type and env specific vars files that load some ec2_*
  overrides
- added the ability to search for amis by ami_name
  - this allows us to specify a base name with a wildcard to have the playbook
    choose the latest available image for that image name
  - added a copy of the ec2_find_ami module that will be in ansible 2.0 until
    we can make ansible 2.0 a requirement.

### CODE REVIEW

**Comment by detiber** [comment]
@wshearn @twiest @tdawson: This change should work better for allowing you guys to override the variables that you need without clobbering the online dev case.

File: filter_plugins/oo_filters.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
 #!/usr/bin/python
 # -*- coding: utf-8 -*-
 # vim: expandtab:tabstop=4:shiftwidth=4
+# pylint: disable=missing-docstring, no-self-use, too-few-public-methods
```
Comment (twiest) [inline filter_plugins/oo_filters.py]: We need a comment saying why these are being disabled and if it's mean to be temporary or permanent.

**Comment by detiber** [comment]
@kwoodson another failure due to ansible missing from the ci host: 
```
+ git/pylint.sh 12ca55504988fe6ea524b222a510d51b0168f95b 5e18c8433733424ff5256e56586bf09970316f15 master
************* Module oo_filters
I:135, 0: Locally disabling too-few-public-methods (R0903) (locally-disabled)
I:135, 0: Locally disabling no-self-use (R0201) (locally-disabled)
F:  8, 0: Unable to import 'ansible' (import-error)
************* Module ec2_ami_find
I:  2, 0: Ignoring entire file (file-ignored)
Build step 'Execute shell' marked build as failure
Setting status of 6792e2c58ea21bd67a36ff4571301782c9f64009 to FAILURE with url https://ci.openshift.redhat.com/jenkins/job/openshift-ansible/22/ and message: Merged build finished.
Finished: FAILURE
```

**Comment by kwoodson** [comment]
@detiber we are moving to the public ci instance.  This was and is working on the old ci but since I do not have access to the new one it is a WIP.  I should have this cleaned up today.

**Comment by twiest** [comment]
@detiber thanks for documenting the pylint disables.

:+1:   I'm going to merge without build bot succeeding since it's broken right now and that's not the fault of this PR.

**Pull request merged**